### PR TITLE
fix link not veritically aligning on android/ios

### PR DESCRIPTION
**Pull request by Justicea83**
### Fixed Issues

$ https://github.com/Expensify/App/issues/8664

### Tests

- [ ] Launch the app on android or ios
- [ ] Login with a public domain email address
- [ ] Click on your profile picture 
- [ ] Click on any workspace > pay bills
- [ ] Link is aligned vertically in pay bills section on public domain

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

- [ ] Launch the app on android or ios
- [ ] Login with a public domain email address(eg. an email that ends in `.gmail.com`)
- [ ] Click on your profile picture 
- [ ] Click on any workspace > pay bills
- [ ] Link is aligned vertically in pay bills section on public domain

#### Web

Screenshot 2022-04-27 at 5 27 42 AM

#### Mobile Web

Screenshot 2022-04-27 at 5 57 27 AM

#### Desktop

Screenshot 2022-04-27 at 5 30 07 AM

#### iOS

Screenshot 2022-04-27 at 5 47 11 AM

#### Android

Screenshot 2022-04-27 at 5 24 01 AM

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -535,7 +535,7 @@ PODS:
     - React-Core
   - RNCMaskedView (0.2.4):
     - React-Core
-  - RNCPicker (1.9.11):
+  - RNCPicker (2.3.1):
```
Comment (parasharrajat) [inline ios/Podfile.lock:538]: Please revert the changes for this file.

File: ios/NewExpensify.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -531,11 +531,11 @@
 			);
 			inputPaths = (
 				"${PODS_ROOT}/Target Support Files/Pods-NewExpensify-NewExpensifyTests/Pods-NewExpensify-NewExpensifyTests-frameworks.sh",
-				"${PODS_XCFRAMEWORKS_BUILD_DIR}/double-conversion/double-conversion.framework/double-conversion",
+				"${PODS_XCFRAMEWORKS_BUILD_DIR}/Flipper-DoubleConversion/double-conversion.framework/double-conversion",
```
Comment (parasharrajat) [inline ios/NewExpensify.xcodeproj/project.pbxproj:534]: Please revert the changes for this file.

File: src/pages/workspace/bills/WorkspaceBillsFirstSection.js [hunk 3]
```diff
@@ -54,11 +56,25 @@ const WorkspaceBillsFirstSection = (props) => {
...
                     {props.user.isFromPublicDomain ? (
-                        <TouchableOpacity
-                            onPress={() => Link.openExternalLink('https://community.expensify.com/discussion/7500/how-to-pay-your-company-bills-in-expensify/')}
-                        >
-                            <Text style={[styles.textBlue]}>example.com@expensify.cash</Text>
-                        </TouchableOpacity>
+                        <>
+                            {
+                                Platform.OS === 'android' || Platform.OS === 'ios'
+                                    ? (
```
Comment (parasharrajat) [inline src/pages/workspace/bills/WorkspaceBillsFirstSection.js:62]: Follow `style.md` guidelines.

File: src/pages/workspace/bills/WorkspaceBillsFirstSection.js [hunk 4]
```diff
@@ -54,11 +56,25 @@ const WorkspaceBillsFirstSection = (props) => {
...
+                                            <Text style={[styles.textBlue]}>example.com@expensify.cash</Text>
+                                        </TouchableNativeFeedback>
+                                    )
+                                    : (
+                                        <TouchableOpacity
+                                            onPress={() => Link.openExternalLink('https://community.expensify.com/discussion/7500/how-to-pay-your-company-bills-in-expensify/')}
+                                        >
+                                            <Text style={[styles.textBlue]}>example.com@expensify.cash</Text>
+                                        </TouchableOpacity>
+                                    )
```
Comment (parasharrajat) [inline src/pages/workspace/bills/WorkspaceBillsFirstSection.js:75]: Why are we using different `Touchable` on different platforms? I do not understand your proposal.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Please fix the link, look at other PRs for reference.
image

- Add QA steps.

File: src/pages/workspace/bills/WorkspaceBillsFirstSection.js (hunk 4 above)
Comment (Justicea83) [reply src/pages/workspace/bills/WorkspaceBillsFirstSection.js:75]: It was voted for in the slack. There TouchableOpacity doesn't work in Android and iOS as expected. You can read more on the links provided in my proposal.

**Comment by Justicea83** [comment]
updated and awaiting another review

File: src/pages/workspace/bills/WorkspaceBillsFirstSection.js [hunk 5]
```diff
@@ -54,11 +56,23 @@ const WorkspaceBillsFirstSection = (props) => {
...
                     {props.translate('workspace.bills.askYourVendorsBeforeEmail')}
                     {props.user.isFromPublicDomain ? (
-                        <TouchableOpacity
-                            onPress={() => Link.openExternalLink('https://community.expensify.com/discussion/7500/how-to-pay-your-company-bills-in-expensify/')}
-                        >
-                            <Text style={[styles.textBlue]}>example.com@expensify.cash</Text>
-                        </TouchableOpacity>
+                        <>
+                            {
+                                Platform.OS === 'android' || Platform.OS === 'ios' ? (
```
Comment (parasharrajat) [inline src/pages/workspace/bills/WorkspaceBillsFirstSection.js:61]: We don't follow this pattern. Use the https://github.com/Expensify/App#platform-specific-file-extensions technique for platform-specific code.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
It seems you missed my last review comments. 

- Add QA steps. 
- Fix the GH_Link from the template.

File: src/pages/workspace/bills/WorkspaceBillsFirstSection.js [hunk 6]
```diff
@@ -54,11 +56,23 @@ const WorkspaceBillsFirstSection = (props) => {
...
+                                    </TouchableNativeFeedback>
+                                ) : (
+                                    <TouchableOpacity
+                                        onPress={() => Link.openExternalLink('https://community.expensify.com/discussion/7500/how-to-pay-your-company-bills-in-expensify/')}
+                                    >
+                                        <Text style={[styles.textBlue]}>example.com@expensify.cash</Text>
+                                    </TouchableOpacity>
+                                )
+                            }
+                        </>
```
Comment (parasharrajat) [inline src/pages/workspace/bills/WorkspaceBillsFirstSection.js:75]: Looking more at the code, I think we should use `TextLink` here.

Check the other links on the app, for example on Connect bank account .
image

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/pages/workspace/bills/WorkspaceBillsFirstSection.js (hunk 6 above)
Comment (parasharrajat) [reply src/pages/workspace/bills/WorkspaceBillsFirstSection.js:75]: More on SignIn page. 
image

Comment (Justicea83) [reply src/pages/workspace/bills/WorkspaceBillsFirstSection.js:75]: There we votes on slack as I said earlier. Your solution was proposed by someone and it was rejected, i can't really tell why. I suggest reading the Proposals in the [issue](https://github.com/Expensify/App/issues/8664).

**Comment by Justicea83** [comment]
cc: @michaelhaxhiu

File: src/pages/workspace/bills/WorkspaceBillsFirstSection.js (hunk 6 above)
Comment (Justicea83) [reply src/pages/workspace/bills/WorkspaceBillsFirstSection.js:75]: cc: @michaelhaxhiu

Comment (parasharrajat) [reply src/pages/workspace/bills/WorkspaceBillsFirstSection.js:75]: Could you please link me to the slack thread?

Comment (stitesExpensify) [reply src/pages/workspace/bills/WorkspaceBillsFirstSection.js:75]: Hi @Justicea83! I'm not seeing a proposal on the issue for using a `<textLink/>` component specifically, only proposals with `<text>` or `<touchableOpacity>`. Am I just missing it somewhere?

Comment (Justicea83) [reply src/pages/workspace/bills/WorkspaceBillsFirstSection.js:75]: You're right, there was no `<textLink/>` in the proposals, but as we had `<touchOpacity/>`, my guts told me to continue with it, at least its tried and tested. I'll try the `<textLink/>` to see what it offers.

Comment (stitesExpensify) [reply src/pages/workspace/bills/WorkspaceBillsFirstSection.js:75]: Sweet, thanks! LMK how it goes

File: src/pages/workspace/bills/WorkspaceBillsFirstSection.js [hunk 7]
```diff
@@ -54,11 +55,11 @@ const WorkspaceBillsFirstSection = (props) => {
                 <Text>
                     {props.translate('workspace.bills.askYourVendorsBeforeEmail')}
                     {props.user.isFromPublicDomain ? (
-                        <TouchableOpacity
-                            onPress={() => Link.openExternalLink('https://community.expensify.com/discussion/7500/how-to-pay-your-company-bills-in-expensify/')}
+                        <TextLink
+                            href="https://community.expensify.com/discussion/7500/how-to-pay-your-company-bills-in-expensify/"
```
Comment (parasharrajat) [inline src/pages/workspace/bills/WorkspaceBillsFirstSection.js:59]: Using `href` here is not correct. It will result in calling the `Link.openUrl` on native. but we want it to be opened via `Link.openExternalLink(`. IMO, `onPress` handler will be best suited.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Thanks for doing the changes.

**Review by parasharrajat — APPROVED** [review]
Looking good. 

cc: @stitesExpensify 

:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by stitesExpensify — APPROVED** [review]
LGTM!
